mne/preprocessing/ica.py

```diff
@@ -2624,3 +2640,53 @@ def corrmap(icas, template, threshold="auto", label=None, ch_type="eeg",
         return template_fig, labelled_ics
     else:
         return None
+
+
+def read_ica_eeglab(fname):
+    """Load ICA information saved in an EEGLAB .set file.
+
+    Parameters
+    ----------
+    fname : str
+        Complete path to a .set EEGLAB file that contains an ICA object.
+
+    Returns
+    -------
+    ica : instance of ICA
+        An ICA object based on the information contained in the input file.
+    """
+    eeg = _check_load_mat(fname, None)
+    info = _get_info(eeg)[0]
+
+    n_components = eeg.icaweights.shape[0]
+
+    ica = ICA(method='imported_eeglab', n_components=n_components)
+
+    ica.current_fit = "eeglab"
+    ica.ch_names = info["ch_names"]
+    ica.n_pca_components = None
+    ica.max_pca_components = n_components
+    ica.n_components_ = n_components
+
+    ica.pre_whitener_ = np.ones((len(eeg.icachansind), 1))
+    ica.pca_mean_ = np.zeros(len(eeg.icachansind))
+
+    n_ch = len(ica.ch_names)
+    assert eeg.icaweights.shape == (n_components, n_ch)
+    if n_components < n_ch:
+        # When PCA reduction is used in EEGLAB, runica returns
+        # weights= weights*sphere*eigenvectors(:,1:ncomps)';
+        # sphere = eye(urchans), so let's use SVD to get our square
+        # weights matrix (u * s) and our PCA vectors (v) back
+        u, s, v = _safe_svd(eeg.icaweights, full_matrices=False)
+        ica.unmixing_matrix_ = u * s
+        ica.pca_components_ = v
+    else:
+        ica.unmixing_matrix_ = eeg.icaweights
+        ica.pca_components_ = eeg.icasphere
+    ica.mixing_matrix_ = linalg.pinv(ica.unmixing_matrix_)
```

Comment:
@christian-oreilly this part I'm not sure about. We no longer dot with `eeg.icawinv` after the `linalg.pinv` here, which you had. But in MNE we should have that the mixing matrix is just the pinv of the unmixing matrix:

https://github.com/mne-tools/mne-python/blob/master/mne/preprocessing/ica.py#L2138

Reply:
I don't remember the exact details here because it was a bit complicated and its been a few month I looked at it now, but in a nutshell, that is what I remember from my conclusion: As far as I understand, there are no exact correspondence between the variables exposed by EEGLAB and MNE for ICA. Futher, after looking for whether EEGLAB exposed variables could be transformed in what MNE is expecting (e.g., through SVD), I was inclined to believed that there were no easy way to do it. So, as far as I know, we will not manage to have something semantically (with respect to the definition of ica.unmixing_matrix and ica.pca_components_ in MNE) perfect here. Given this limitation, my efforts were directed at having an import function that gives numerically the correct output when the ICA is applied in MNE versus when applied in EEGLAB; I validated this by running tests. One of these tests is part of this PR.